BrainPortal/app/controllers/data_providers_controller.rb

```diff
@@ -779,20 +779,33 @@ def repair
     @provider = DataProvider.find(params[:id])
     @issues   = @provider.provider_report.select { |i| params[:issue_ids].include? i[:id].to_s }
 
-    # Try to repair the inconsistencies (or issues)
-    failed_list  = []
-    success_list = []
-    @issues.each do |issue|
-      begin
-        @provider.provider_repair(issue)
-        success_list << issue
-      rescue => ex
-        failed_list  << [issue, ex]
+    if @issues.blank?
+      respond_to do |format|
+        format.html { redirect_to :action => :report }
       end
+      return
     end
 
-    # Display a message reporting how many issues were repaired
-    unless @issues.empty?
+    Message.send_message(current_user,
+      :message_type  => :notice,
+      :header        => "Repair #{@issues.size} inconsistencies in background.",
+    )
+
+    CBRAIN.spawn_with_active_records(:admin, "Repair DP inconsistencies") do
+      # Try to repair the inconsistencies (or issues)
+      failed_list  = []
+      success_list = []
+
+      @issues.each do |issue|
```

Comment:
Change to `each_with_index_and_size` then within loop add

```ruby
Process.setproctitle "Repair DP=#{@provider.id} #{idx+1}/#{size}"
```

(see other examples in the same controller)